### DISCUSSION TITLE
Adding a "generic-hmac" notification Provider type

### DISCUSSION THREAD

**Discussion by makkes** [fluxcd/flux2#3159] [Proposals]
In order to be able to authenticate events sent by notification-controller to an external generic webhook, I'd like to discuss adding an additional Provider type "generic-hmac" that works exactly like the "generic" provider but with the addition that it consumes a key from a Secret, creates an HMAC for the payload and adds an "X-Signature" header to the request carrying the HMAC.

The implementation is very simple, I have a prototype running locally. We could also get away without adding an additional field to be consumed from `.spec.secretRef` by re-using one of the existing ones, e.g. "password" or "token".

**Comment by stefanprodan** [comment]
More details on how this could look like here: https://github.com/fluxcd/notification-controller/issues/99#issuecomment-1262142759
